library/ssl_tls.c

```diff
@@ -8337,8 +8337,16 @@ int mbedtls_ssl_get_record_expansion( const mbedtls_ssl_context *ssl )
     const mbedtls_ssl_transform *transform = ssl->transform_out;
     unsigned block_size;
 
+    size_t in_hdr_len, out_hdr_len, max_hdr_len;
+    in_hdr_len = mbedtls_ssl_in_hdr_len( ssl );
+    out_hdr_len = mbedtls_ssl_out_hdr_len( ssl );
+    if( in_hdr_len < out_hdr_len )
+        max_hdr_len = out_hdr_len;
+    else
+        max_hdr_len = in_hdr_len;
```

Comment:
I understand the intention but I'm not convinced this is the best thing to do.

Currently the documentation is ambiguous, as it does not mention explicitly if this is only about outgoing records or also incoming ones, but the current implementation is only about outgoing (as for CBC we count one 1 block of padding, while incoming records emitted by a different implementation could have up to 256 bytes of padding), and I think the intended use is also quite clearly for outgoing records (I have trouble coming up with a use case where you need to know the maximum expansion for incoming records).

So I'd be inclined to just use `out_hdr_len` here and clarify the documentation.

Reply:
@mpg I wasn't sure here either, only noticed that the documentation was ambiguous, and decided to go with the maximum as a safe overapproximation. 

> So I'd be inclined to just use out_hdr_len here and clarify the documentation.

I'm ok with that.